nemo/collections/llm/gpt/model/megatron/hyena/hyena_utils.py

```diff
@@ -507,9 +521,13 @@ def __init__(
             gamma = gamma.cuda().log()
             self.gamma = nn.Parameter(gamma)
 
-            R = 1e-1 * torch.randn(d_model, order, dtype=torch.float32) / math.sqrt(order)
+            R = (
+                1e-1
+                * torch.randn(d_model, order, dtype=torch.float32, device=torch.cuda.current_device())
+                / math.sqrt(order)
+            )
             self.R = nn.Parameter(R)
-            self.p = nn.Parameter(-torch.ones(d_model, order, dtype=torch.float32))
+            self.p = nn.Parameter(-torch.ones(d_model, order, dtype=torch.float32, device=torch.cuda.current_device()))
```

Comment:
I don't love setting the device like this here. This relies on the env having properly set current_device etc. What if you wanted to initialize on meta or something? Please either remove or add a comment about what is being fixed. If there's another tensor in context that you want to do a "same device" placement then I would recommend a ones like or an `other.device`.